lib/checkother.cpp

```diff
@@ -3105,3 +3105,20 @@ void CheckOther::comparePointersError(const Token *tok, const ValueFlow::Value *
     reportError(
         errorPath, Severity::error, "comparePointers", verb + " pointers that point to different objects", CWE570, false);
 }
+
+void CheckOther::checkModuloOfOne()
+{
+    for (const Token *tok = mTokenizer->tokens(); tok; tok = tok->next()) {
+        if (!tok->astOperand2() || !tok->astOperand1())
+            continue;
+        if (tok->str() != "%")
+            continue;
+        if (!tok->valueType() || !tok->valueType()->isIntegral())
+            continue;
+
+        // Value flow..
+        const ValueFlow::Value *value = tok->astOperand2()->getValue(1LL);
+        if (value)
```

Comment:
It seems to me we should only warn if the rhs is always `1`. Not if it can be `1`. Test case:

    void foo() {
        for (int x = 1; x < 10; ++x) {
            int y = 100 % x;
        }
    }

You can change this condition to:

    if (value && value->isKnown())